### ISSUE TITLE
[REGRESSION]:

### ISSUE THREAD

**Issue by a60915** [microsoft/azure-pipelines-tasks#20581]
### New issue checklist

- [x] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [ ] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [ ] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

AzurePowerShell

### Breaking task version

5.247.5

### Last working task version

5.246.1

### Regression Description

Attempting to run an inline script using AzurePowerShell generates the following error message:

```
An error occurred in Initialize-AzModule
##[error]The term 'Resolve-Error' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
##[error]PowerShell exited with code '1'.
Finishing: Download from Azure

```

### Environment type (Please select at least one enviroment where you face this issue)

- [x] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Windows Server 2019 Datacenter

### Relevant log output

An error occurred in Initialize-AzModule
##[error]The term 'Resolve-Error' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
##[error]PowerShell exited with code '1'.
Finishing: Download from Azure

### Full task logs with system.debug enabled

<details>
  <summary>UNSUCCESSFUL RUN</summary>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

<details>
  <summary>SUCCESSFUL RUN</summary>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

**Comment by v-schhabra** [comment]
Hi @a60915 
Could you pls share the complete pipeline debug logs by adding variable system.debug to "true".

**Comment by v-schhabra** [comment]
Just wanted to update that error is not reproducible at our end.

**Comment by v-schhabra** [comment]
@a60915 Thanks for sharing the logs. 
Could you pls share the inline script as well?

**Comment by a60915** [comment]
@v-schhabra Here you go:

```powershell
Write-Host "Downloading file from Azure..."

# Get the storage account
$storageContext = New-AzStorageContext -StorageAccountName "prodwbint1file1" -StorageAccountKey "$(StorageAccount.prodwbint1file1.AccessKey)"

# Get the file
Get-AzStorageBlobContent -Container "dealerupload" -Blob "businessdirectoryxml.xml" -Destination ".\" -Context $storageContext -Force
```

**Comment by v-schhabra** [comment]
@a60915 Did you configured a new agent or any changes you did recently?

**Comment by a60915** [comment]
@v-schhabra No, nothing changed on the agent. I did see there was a recent change to the task, adding the call to Resolve-Error (#20528) . I'm guessing this is in a module that isn't installed by default, but you are relying on it being there.

**Comment by v-schhabra** [comment]
@a60915 Could you pls tell which PowerShell version you are using?

**Comment by a60915** [comment]
@v-schhabra 
```
Name                           Value
----                           -----
PSVersion                      5.1.17763.6414
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17763.6414
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by jngibert** [comment]
Also seeing this in our environment too.

A temp workaround is to downgrade the Az.Accounts module to 3.0.5, but that has it's own side effects.

Agree that the catch block's use of the `Resolve-Error` alias in **Tasks/AzurePowerShellV5/CoreAz.ps1** should be corrected to `Resolve-AzError`

**Comment by ghdgnsvy** [comment]
Same exact problem for me as well. Is there any plan on fixing this? This issue has been open for months now.

**Comment by thebeautiful** [comment]
Another month has passed with no fix, just letting you know :-)

**Comment by svdk01** [comment]
Same problem here, we have multiple pools deployed where most pools are working correctly but 1 pool has this specific problem.

**Comment by SoucianceEqdamRashti** [comment]
We faced this issue today. When will a fix be made available? It makes self-hosted agents a pain to maintain.

**Comment by anujbansal-ms** [comment]
any update on how to resolve this issue

**Comment by zeilemaker54** [comment]
Why does it take so long to implement such a simple change ? Someone actually has made a pull request (https://github.com/microsoft/azure-pipelines-tasks/pull/20955)
Customers are warned beforehand that Resolve-Error will be removed, but Microsoft seemes to 'forget' to update their own code.
Even when an issue is raised, nothing happens for months. I am stuned....

**Comment by lzandman** [comment]
> Why does it take so long to implement such a simple change ? Someone actually has made a pull request ([#20955](https://github.com/microsoft/azure-pipelines-tasks/pull/20955)) Customers are warned beforehand that Resolve-Error will be removed, but Microsoft seemes to 'forget' to update their own code. Even when an issue is raised, nothing happens for months. I am stuned....

I fully agree with @zeilemaker54. Unbelievable!

**Comment by JoshHariz** [comment]
Still broken 9 months later? When its a single line change? This is unbelievable

**Comment by HamsterExAstris** [comment]
It's more than just a single-line change. Using `Resolve-AzError` fixes the issue when the Az module is installed, but users who try to run the task on a machine without the module will encounter the same "not recognized as the name of a cmdlet, function, script file, or operable program" error. It needs to be fixed to not rely on that method in the Az module at all, so that `Initialize-AzModule`'s message about the module being missing is reported to the user.

**Comment by zeilemaker54** [comment]
> It's more than just a single-line change. Using `Resolve-AzError` fixes the issue when the Az module is installed, but users who try to run the task on a machine without the module will encounter the same "not recognized as the name of a cmdlet, function, script file, or operable program" error. It needs to be fixed to not rely on that method in the Az module at all, so that `Initialize-AzModule`'s message about the module being missing is reported to the user.

Replacing Resolve-Error by Resolve-AzError will not change the behaviour, so the problem with Resolve-AzError not being available when the Az module is not loaded is no different then Resolve-Error. And further more, I would expect that after 9 months someone can come up with a solution to detect this situation and act upon it.
By not solving the issue, you effectively can not use the Az 14.x.x powershell module with the Azure Pipelines Tasks extensions. I think this is unacceptable.

**Comment by lzandman** [comment]
Maybe the team that would be responsible for this has been let go during the last rounds of mass Microsoft layoffs?

**Comment by v-schhabra** [comment]
Hi Everyone,
Sorry for the inconvenience caused due to this issue. We are investigating on this and will keep here posted on the updates.

**Comment by v-schhabra** [comment]
Deployments are completed for this fix. Closing the WI.
